src/apps/orders/services.py

```diff
@@ -0,0 +1,127 @@
+            if not len(set(product_ids)) == len(products):
+                raise ServiceException("Wrong products!")
+
+        order_create_data = dict()
+        order_create_data["user"], order_create_data["products"] = user, products
+        new_order = Order(**order_create_data)
+        session.add(new_order)
+        session.commit()
+
+    return OrderOutputSchema.from_orm(new_order)
+
+
+def get_single_order(session: Session, order_id: int) -> OrderOutputSchema:
+    if not (order_object := if_exists(Order, "id", order_id, session)):
+        raise DoesNotExist(Order.__name__, order_id)
+
+    return OrderOutputSchema.from_orm(order_object)
+
+
+def get_all_orders(
+    session: Session, page_params: PageParams
+) -> PagedResponseSchema[OrderOutputSchema]:
+    query = select(Order)
+
+    return paginate(
+        query=query,
+        response_schema=OrderOutputSchema,
+        table=Order,
+        page_params=page_params,
+        session=session,
+    )
+
+
+def get_all_user_orders(
+    session: Session, user_id: int, page_params: PageParams
+) -> PagedResponseSchema[OrderOutputSchema]:
+    query = select(Order).filter(Order.user_id == user_id)
+
+    return paginate(
+        query=query,
+        response_schema=OrderOutputSchema,
+        table=Order,
+        page_params=page_params,
+        session=session,
+    )
+
+
+def update_single_order(
+    session: Session, order_input: OrderInputSchema, order_id: int
+) -> OrderOutputSchema:
+    if not (order_object := if_exists(Order, "id", order_id, session)):
+        raise DoesNotExist(Order.__name__, order_id)
+
+    order_data = order_input.dict(exclude_none=True, exclude_unset=True)
+    if order_data:
+        if order_data.get("product_ids"):
+            incoming_products = set(order_data["product_ids"])
+            current_products = set(product.id for product in order_object.products)
+
+            if to_delete := current_products - incoming_products:
+                session.execute(
+                    delete(order_product_association_table).where(
+                        Product.id.in_(to_delete)
+                    )
+                )
+
+            if to_insert := incoming_products - current_products:
+                rows = [
+                    {"order_id": order_id, "product_id": product_id}
+                    for product_id in to_insert
+                ]
+                session.execute(insert(order_product_association_table).values(rows))
+
+            order_data.pop("product_ids")
+
+        statement = update(Order).filter(Order.id == order_id).values(**order_data)
+
+        session.execute(statement)
+        session.commit()
+        session.refresh(order_object)
+
+    return get_single_order(session, order_id=order_id)
+
+
+def delete_all_orders(session: Session):
+    statement = delete(Order)
+    result = session.execute(statement)
+    session.commit()
+
+    return result
+
+
+def delete_single_order(session: Session, order_id: int):
```

Comment:
Predzej anulowane niz usuniete, bedzie w wielu miejscach apki wymagana dodatkowa logika co do walidacji produktow

Reply:
może i tak, bo do historii zamówień się przyda, na razie to usunę, a potem w next PR jak będą statusy to rozbuduję to